packages/aws-cdk-lib/aws-cognito/lib/user-pool-group.ts

```diff
@@ -0,0 +1,120 @@
+import { Construct } from 'constructs';
+import { CfnUserPoolGroup } from './cognito.generated';
+import { IUserPool } from './user-pool';
+import { IRole } from '../../aws-iam';
+import { IResource, Resource, Token } from '../../core';
+
+/**
+ * Represents a user pool group.
+ */
+export interface IUserPoolGroup extends IResource {
```

Comment:
@mazyu36 thanks for this PR - I'm curious why you added this base interface and if it is necessary?

Reply:
This follows the design guidelines. 
I recognize that Interfaces are essential as they are implemented across almost all resources.

https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#construct-interface